picard/coverart/providers/caa.py

```diff
@@ -46,6 +46,61 @@
 
 _CAA_IMAGE_SIZE_DEFAULT = 500
 
+_CAA_IMAGE_TYPE_DEFAULT_INCLUDE = [ 'front', ]
+_CAA_IMAGE_TYPE_DEFAULT_EXCLUDE = [ 'raw/unedited', 'watermark', ]
+
+
+def make_list_box():
+    """Make standard list box for CAA image type selection dialog."""
+    list_box = QtWidgets.QListWidget()
+    list_box.setFixedSize(QtCore.QSize(150, 250))
+    list_box.setSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
+    list_box.setSortingEnabled(True)
+    list_box.setSelectionMode(QtWidgets.QAbstractItemView.ExtendedSelection)
+    return list_box
+
+def make_arrow_button(label, command):
+    """Make standard arrow button for CAA image type selection dialog."""
+    item = QtWidgets.QPushButton(label)
+    item.clicked.connect(command)
+    item.setFixedSize(QtCore.QSize(35, 20))
+    return item
+
+def make_arrows_column(list_add, list_add_all, list_remove, list_remove_all, reverse=False):
+    """Make standard arrow buttons column for CAA image type selection dialog."""
+    _spacer_item = QtWidgets.QSpacerItem(20, 20, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
+    arrows = QtWidgets.QWidget()
+    arrows_layout = QtWidgets.QVBoxLayout()
+    arrows.setLayout(arrows_layout)
+    arrows_layout.addItem(QtWidgets.QSpacerItem(_spacer_item))
+    arrows_layout.addWidget(make_arrow_button('>' if reverse else '<', list_add))
+    arrows_layout.addWidget(make_arrow_button('>>' if reverse else '<<', list_add_all))
+    arrows_layout.addWidget(make_arrow_button('<' if reverse else '>', list_remove))
+    arrows_layout.addWidget(make_arrow_button('<<' if reverse else '>>', list_remove_all))
+    arrows_layout.addItem(QtWidgets.QSpacerItem(_spacer_item))
+    return arrows
+
+def move_item(item, list1, list2):
+    """Move the specified item from one listbox to another.  Used in the CAA image type selection dialog."""
+    clone = item.clone()
+    list2.addItem(clone)
+    list1.takeItem(list1.row(item))
```

Comment:
Perhaps you can skip the `clone()` if you remove it first from list1 then add it to list2:

```python
list2.addItem(list1.takeItem(list1.row(item)))
```

Untested though.

Reply:
It works, but I had to remove it from the source list before I could add it to the target list.